### PR TITLE
Region Docs #2959

### PR DESCRIPTION
**Pull request by scott-w**
I’ve started chopping up the region docs to be a bit more detailed on
regions and the different ways of using them.

I'll need some help with the following:

1. What events get fired and where
2. Whether `Region` is deprecated in public docs

If `Region` isn't deprecated, I feel we should move it to the bottom as an "advanced" topic.

### CODE REVIEW

**Comment by jasonLaster** [comment]
👍 

I really like changing the titles from the method to the purpose

File: docs/marionette.region.md [hunk 1]
```diff
@@ -74,183 +74,87 @@ var MyView = Marionette.View.extend({
...
 
-var MyView = Marionette.View.extend({
-  regions: {
-    contentRegion: {
-      regionClass: MyRegion,
-      myRegionOption: 'bar',
-      myOtherRegionOption: 'baz'
-    }    
+  onRender: function() {
+    this.showChildView('tooverwrite', new OverWriteView());
```
Comment (rafde) [inline docs/marionette.region.md]: 😕 Shouldn't this be `this.showChildView('main', new OverWriteView());`
Or should the `main` key be renamed `tooverwrite` in `regions`?

Comment (scott-w) [inline docs/marionette.region.md]: Your first comment is correct - it was an oversight on my part.

**Comment by ahumphreys87** [comment]
Really great work :+1:

**Comment by denar90** [comment]
Thanks @scott-w 👍

**Pull request merged**